diag_manager/fms_diag_output_buffer.F90

```diff
@@ -72,6 +73,11 @@ module fms_diag_output_buffer_mod
   procedure :: get_field_id
   procedure :: set_yaml_id
   procedure :: get_yaml_id
+  procedure :: write_buffer
+  !! These are needed because otherwise the write_data calls will go into the wrong interface
+  procedure :: write_buffer_wrapper_netcdf
+  procedure :: write_buffer_wrapper_domain
+  procedure :: write_buffer_wrapper_u
```

Comment:
Why isn't `write_data` and interface and the three subroutines `module procedure`s in the interface?

Reply:
I need the `write buffer` routine because i need the select types for the fileobjs so it can go in the correct interface and because `write_buffer` is called a few times it will be messy without it. 

I couldn't get the interface to work because these routines are part of the object and not the module.